tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -139,22 +194,27 @@ private static double ConvertTimestamp(long ns)
             return ns << shift;
         }
 
-        private static StatsAggregationKey BuildKey(Span span)
+        private async Task InitializeAsync()
         {
-            var rawHttpStatusCode = span.GetTag(Tags.HttpStatusCode);
+            try
+            {
+                var isDiscoverySuccessful = await _discoveryService.DiscoverAsync().ConfigureAwait(false);
```

Comment:
As you've probably noticed, the way we're currently using `DiscoveryService` where we're only querying it on initialization means that if the customer enables stats in `datadog-agent` config (or upgrades `datadog-agent` to a version that has these endpoints) **after** the monitored processes have started, they'd need to restart all the monitored processes for the change to take effect. 

This is not an ideal user experience, so I think we should definitely consider changing things such that `DiscoveryService` such that it's continuously polling the `/info` endpoint and triggering some sort of event / notification to all of its consumers when things change.

This is just a general comment - not related to your PR really, just wanted to bring it to your attention.

Reply:
This continuous check would benefit this feature as well, because if the agent has been upgraded/downgrade while the application is alive, we should adjust in real-time